src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -2140,3 +2146,93 @@ export const handleUpdateCollaborators = [
   }),
   _handleUpdateCollaborators,
 ] as RequestHandler[]
+
+/**
+ * NOTE: Exported for testing.
+ * Private handler for PUT /forms/:formId/start-page
+ * @precondition Must be preceded by request validation
+ * @security session
+ *
+ * @returns 200 with updated start page
+ * @returns 403 when current user does not have permissions to update the start page
+ * @returns 404 when form cannot be found
+ * @returns 410 when updating the start page for an archived form
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const _handleUpdateStartPage: RequestHandler<
+  { formId: string },
+  IFormDocument['startPage'] | ErrorDto,
+  StartPageUpdateDto
+> = (req, res) => {
+  const { formId } = req.params
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+
+  // Step 1: Retrieve currently logged in user.
+  return (
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Retrieve form with write permission check.
+        AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Write,
+        }),
+      )
+      // Step 3: User has permissions, proceed to allow updating of start page
+      .andThen(() => AdminFormService.updateStartPage(formId, req.body))
+      .map((updatedStartPage) =>
+        res.status(StatusCodes.OK).json(updatedStartPage),
+      )
+      .mapErr((error) => {
+        logger.error({
+          message: 'Error occurred when updating start page',
+          meta: {
+            action: '_handleUpdateStartPage',
+            ...createReqMeta(req),
+            userId: sessionUserId,
+            formId,
+            body: req.body,
+          },
+          error,
+        })
+        const { errorMessage, statusCode } = mapRouteError(error)
+        return res.status(statusCode).json({ message: errorMessage })
+      })
+  )
+}
+
+/**
+ * Handler for PUT /forms/:formId/start-page
+ */
+export const handleUpdateStartPage = [
+  celebrate({
+    [Segments.BODY]: {
+      paragraph: Joi.string().allow('').optional(),
+      estTimeTaken: Joi.number().min(1).max(1000).required(),
+      colorTheme: Joi.string()
+        .valid(...Object.values(Colors))
+        .required(),
+      logo: Joi.object({
+        state: Joi.string().valid(...Object.values(FormLogoState)),
+        fileId: Joi.when('state', {
+          is: FormLogoState.Custom,
+          then: Joi.string().required(),
```

Comment:
oh dang didn't know about this; i thought it would default to the normal joi behaviour where it rejects extra keys @__@ TIL and i'll update this (and below)

Reply:
Idk actually, you can experiment!

Reply:
you're right - need the condition to reject :(